### ISSUE TITLE
Improperly formed Table object can cause infinite loop

### ISSUE THREAD

**Issue by rraustad** [NREL/EnergyPlus#5234]
HelpDesk Ticket 10801

User found a crash with an improperly formed Table:TwoIndependentVariables object. Running the defect file in V8.4 shows an infinite loop in CurveManager at or around line 2080. Additional checks on input data is necessary to make this more robust.

MinTableData = 999999.0;
MinTableData = minval( TempTableData( TableNum ).X1 );
for ( VarIndex = 1; VarIndex <= MaxTableNums; ++VarIndex ) {
    if ( TempTableData( TableNum ).X1( VarIndex ) == MinTableData ) {
        TableData( TableNum ).X1( NumXVar ) = TempTableData( TableNum ).X1( VarIndex );
        TableData( TableNum ).X2( NumXVar ) = TempTableData( TableNum ).X2( VarIndex );
        TableData( TableNum ).Y( NumXVar ) = TempTableData( TableNum ).Y( VarIndex );
        TempTableData( TableNum ).X1( VarIndex ) = 999999.0;
        ++NumXVar;
    }
}

Expect this error message when the issue is resolved:

```
** Severe  ** GetCurveInput: For Table:TwoIndependentVariables: AWHP_RCAP
**   ~~~   ** The number of data entries must be evenly divisable by 3. Number of data entries = 134
** Severe  ** GetCurveInput: For Table:TwoIndependentVariables: AWHP_RCOP
**   ~~~   ** The number of data entries must be evenly divisable by 3. Number of data entries = 134
**  Fatal  ** GetCurveInput: Errors found in getting Curve Objects.  Preceding condition(s) cause termination.
```

**Comment by JasonGlazer** [comment]
@jasondegraw I was considering working on this bug soon but I heard today that you were refactoring some related code. Is what you are working on likely to impact the code related to this?

**Comment by jasondegraw** [comment]
@JasonGlazer I am working on refactoring the curves and tables, but first I am writing unit tests here:

https://github.com/NREL/EnergyPlus/blob/add-curve-unit-tests/tst/EnergyPlus/unit/Curves.unit.cc

and here:

https://github.com/NREL/EnergyPlus/blob/add-curve-unit-tests/tst/EnergyPlus/unit/Tables.unit.cc

It has not gone so well with the tables. If you have bandwidth to help with my unit testing problems that would be great, but definitely go ahead with the bug fix. Refactoring working code is better and easier.

**Comment by JasonGlazer** [comment]
@rraustad I just ran an updated file for this bug and got the exact error message you said should occur when the issue was resolved:

```
Program Version,EnergyPlus, Version 8.5.0-c87e61b44b, YMD=2016.05.19 12:25,IDD_Version 8.5.0
   ** Severe  ** GetCurveInput: For Table:TwoIndependentVariables: AWHP_RCAP
   **   ~~~   ** The number of data entries must be evenly divisable by 3. Number of data entries = 134
   ** Severe  ** GetCurveInput: For Table:TwoIndependentVariables: AWHP_RCOP
   **   ~~~   ** The number of data entries must be evenly divisable by 3. Number of data entries = 134
   **  Fatal  ** GetCurveInput: Errors found in getting Curve Objects.  Preceding condition(s) cause termination.
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=2
   ..... Last severe error=GetCurveInput: For Table:TwoIndependentVariables: AWHP_RCOP
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 0 Warning; 2 Severe Errors; Elapsed Time=00hr 00min  0.56sec
```

Is it possible you fixed this bug already?

**Comment by rraustad** [comment]
No, I haven't done anything with this one. I think I would debug and break at the point where I last saw an infinite loop to see if the table data shows something funky. This could still cause an issue in the future unless something corrects an ill formed table before this loop. I see two places now and can't recall which one was stuck in the infinite loop. It could be that @DeadParrot changed something to catch the problem before these loops.

https://github.com/NREL/EnergyPlus/blame/add-curve-unit-tests/src/EnergyPlus/CurveManager.cc#L1947

https://github.com/NREL/EnergyPlus/blame/add-curve-unit-tests/src/EnergyPlus/CurveManager.cc#L2135

**Comment by JasonGlazer** [comment]
I was able to reproduce the problem in 8.5.0. I updated the defect file. The infinite loop in in the second group. It looks like it may be related to a NaN.

**Comment by mjwitte** [comment]
Closed via #5674
